src/main/java/org/opensearch/neuralsearch/processor/factory/RerankProcessorFactory.java

```diff
@@ -37,6 +38,7 @@ public class RerankProcessorFactory implements Processor.Factory<SearchResponseP
     public static final String CONTEXT_CONFIG_FIELD = "context";
 
     private final MLCommonsClientAccessor clientAccessor;
+    private final Environment environment;
```

Comment:
Please check if you can dynamically update the plugins.neural_search.reranker_max_document_fields. 
I think you need to use `ClusterSetting` rather than environment. Environment only reads setting from opensearch.yml file.

Reply:
ack